frontend/pages/hosts/ManageHostsPage/components/RunScriptBatchModal/RunScriptBatchModal.tsx

```diff
@@ -68,14 +77,17 @@ const RunScriptBatchModal = ({
   const onRunScriptBatch = useCallback(
     async (script: IScript) => {
       setIsUpdating(true);
+      const body = runByFilters
+        ? // satisfy IScriptBatchSupportedFilters
+          { script_id: script.id, filters: { ...filters, team_id: teamId } }
+        : { script_id: script.id, host_ids: selectedHostIds };
       try {
-        await scriptsAPI.runScriptBatch({
-          host_ids: selectedHostIds,
-          script_id: script.id,
-        });
+        await scriptsAPI.runScriptBatch(body);
         renderFlash(
           "success",
-          `Script is running on ${selectedHostIds.length} hosts, or will run as each host comes online. See host details for individual results.`
+          `Script is running on ${
+            runByFilters ? totalFilteredHostsCount : selectedHostIds.length
```

Comment:
use `toLocaleString()` here and in the "will run on" text so that we get commas? Currently it's:

---

<img width="575" alt="image" src="https://github.com/user-attachments/assets/a7104879-e3d9-4928-8da5-4e4f75aeae9d" />